File: Backend/Data/Entities/Discipline.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Data.Entities
{
    public class Discipline
    {
        [Required]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }

        protected bool Equals(Discipline other)
        {
            return Id == other.Id;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Discipline) obj);
        }

        public override int GetHashCode()
        {
            return Id;
        }
    }
}

File: Android/app/src/main/java/com/khsm/app/data/entities/Discipline.java
package com.khsm.app.data.entities;

import java.io.Serializable;

public class Discipline implements Serializable{
    public final Integer id;
    public final String name;
    public final String description;

    public Discipline(Integer id, String name, String description){
        this.id = id;
        this.name = name;
        this.description = description;
    }
}
